system-helper/flatpak-system-helper.c

```diff
@@ -237,10 +332,53 @@ handle_deploy (FlatpakSystemHelper   *object,
   gboolean local_pull;
   gboolean reinstall;
   g_autofree char *url = NULL;
+  g_autoptr(OngoingPull) ongoing_pull = NULL;
+  g_autofree gchar *repo_path = NULL;
+  gint dir_fd = -1;
 
-  g_debug ("Deploy %s %u %s %s %s", arg_repo_path, arg_flags, arg_ref, arg_origin, arg_installation);
+  ongoing_pull = take_ongoing_pull_by_dir (arg_repo_path);
+  if (ongoing_pull != NULL)
+    {
+      g_autofree gchar *repo_basename = NULL;
+      g_autoptr(GError) local_error = NULL;
 
-  system = dir_get_system (arg_installation, get_sender_pid (invocation), &error);
+      if (umount (ongoing_pull->mountpoint) == -1)
+        {
+          /* TODO: What else should do we here, other than error-reporting */
+          glnx_throw_errno_prefix (&local_error, "Failed to unmount dir %s", ongoing_pull->mountpoint);
+          g_warning ("%s", local_error->message);
+          g_clear_error (&local_error);
+        }
+
+      /* Killing the fuse backend will revoke all access to write operations. */
+      g_subprocess_force_exit (ongoing_pull->revokefs_backend);
+      g_clear_object (&ongoing_pull->revokefs_backend);
+
+      repo_basename = g_path_get_basename (arg_repo_path);
+      repo_path = g_build_filename (ongoing_pull->src_dir, repo_basename, NULL);
+      repo_file = g_file_new_for_path (repo_path);
+
+      dir_fd = open (ongoing_pull->src_dir, O_DIRECTORY);
+      if (!flatpak_canonicalize_permissions (dir_fd,
+                                             repo_path,
+                                             getuid() == 0 ? 0 : -1,
+                                             getuid() == 0 ? 0 : -1,
+                                             &local_error))
+        {
+          g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                                                 "Failed to canoncalize permissions: %s", local_error->message);
```

Comment:
Should the repo be cleaned up in this case?

Reply:
Yes. It will be cleared up by the `g_autoptr` of ongoing_pull